src/mixins/filing-template-mixin.ts

```diff
@@ -530,18 +546,39 @@ export default class FilingTemplateMixin extends DateMixin {
     this.setFoundingDate(draftFiling.business.foundingDate)
 
     // restore Restoration data
+    if (draftFiling.restoration.applicationDate) {
+      this.setRestorationApplicationDate(draftFiling.restoration.applicationDate)
+    }
+    this.setRestorationApprovalType(draftFiling.restoration.approvalType)
+    if (draftFiling.restoration.courtOrder) {
+      this.setRestorationCourtOrder(draftFiling.restoration.courtOrder)
+    }
     this.setRestorationType(draftFiling.restoration.type)
-    this.setRestorationExpiry(draftFiling.restoration.expiry || null)
-    this.setRestorationRelationships(draftFiling.restoration.relationships || [])
+    if (draftFiling.restoration.expiry) {
+      this.setRestorationExpiry(draftFiling.restoration.expiry)
+    }
+    if (draftFiling.restoration.relationships) {
+      this.setRestorationRelationships(draftFiling.restoration.relationships)
+    }
+    if (draftFiling.restoration.noticeDate) {
+      this.setRestorationNoticeDate(draftFiling.restoration.noticeDate)
+    }
+    if (draftFiling.restoration.noticeDate) {
+      this.setRestorationApplicationDate(draftFiling.restoration.noticeDate)
+    }
```

Comment:
Is there an opportunity to separate the code associated with each filingType?  I don't know if this would work, but I'd like to try diverting the logic by filingType early so we can simplify our code and reduce the number of times where we have to write a series of `if-then`, `if-then-else` or `switch` statements.

Reply:
Yes but not the code you attached your comment to. This particular code does a conditional restore of each field, so it doesn't overwrite the initial store value if the draft filing (JSON) doesn't contain it.

Reply:
@jonathan-longe OK to resolve this conversation?